src/main/cpp/RLBotInterface/BallPrediction/BallPrediction.cpp

```diff
@@ -0,0 +1,126 @@
+// BallPrediction.cpp : Defines the entry point for the console application.
+//
+
+#include <stdio.h>
+#include <chrono>
+#include <thread>
+#include <csignal>
+#include "PredictionService.hpp"
+#include <GameFunctions\GameFunctions.hpp>
+#include <rlbot_generated.h>
+#include <BoostUtilities\BoostUtilities.hpp>
+#include <BoostUtilities\BoostConstants.hpp>
+
+
+vec3 convertVec(const rlbot::flat::Vector3* vec)
+{
+	return vec3{ vec->x(), vec->y(), vec->z() };
+}
+
+rlbot::flat::Vector3 convertVec(vec3 vec)
+{
+	return rlbot::flat::Vector3(vec[0], vec[1], vec[2]);
+}
+
+void emplacePrediction(std::list<BallPrediction::BallSlice>* prediction) {
+
+	std::vector<flatbuffers::Offset<rlbot::flat::PredictionSlice>> slices;
+
+	flatbuffers::FlatBufferBuilder builder;
+
+	for (std::list<BallPrediction::BallSlice>::iterator it = prediction->begin(); it != prediction->end(); it++) {
+
+		rlbot::flat::PhysicsBuilder physicsBuilder(builder);
+		physicsBuilder.add_location(&convertVec(it->Location));
+		physicsBuilder.add_velocity(&convertVec(it->Velocity));
+		physicsBuilder.add_angularVelocity(&convertVec(it->AngularVelocity));
+
+		auto physOffset = physicsBuilder.Finish();
+
+		rlbot::flat::PredictionSliceBuilder sliceBuilder(builder);
+		sliceBuilder.add_gameSeconds(it->gameSeconds);
+		sliceBuilder.add_physics(physOffset);
+		slices.push_back(sliceBuilder.Finish());
+	}
+
+	auto slicesOffset = builder.CreateVector(slices);
+
+	rlbot::flat::BallPredictionBuilder predictionBuilder(builder);
+	predictionBuilder.add_slices(slicesOffset);
+
+	builder.Finish(predictionBuilder.Finish());
+
+	static BoostUtilities::SharedMemWriter ballPredictionWriter(BoostConstants::BallPredictionName);
+
+	ballPredictionWriter.writeData(builder.GetBufferPointer(), builder.GetSize());
+}
+
+
+int runBallPrediction()
+{
+	printf("Ball Prediction Service Started\n");
+
+	BallPrediction::PredictionService predictionService(6.0, 1.0 / 60);
```

Comment:
can we add a comment for what these primitives are?
something like
/* field_name */ 6.0, /* field_name */ 1.0